Improve documentation in basis group (#208)
TREX-CoE/trexio: trex.org

  supported by the client software. For this reason, normalization
  constants are explicitly provided in the TREXIO files to
  ensure portability and completeness.

  We consider here basis functions centered on nuclei. It is
  therefore possible to introduce *dummy atoms* (artificial centers
  with no physical meaning) solely for the purpose of placing basis
  functions at arbitrary locations in space.

  The atomic basis set is defined as a list of shells. Each shell
  $s$ is centered on a point $\mathbf{R}_A$, associated with a given
  angular momentum $\ell_s$, and characterized by a radial function
  $R_s(\mathbf{r})$. Note that since the angular functions are
  defined as regular solid harmonics that include the $r^{\ell_s}$ factor,
  the radial functions in TREXIO differ from the textbook definition
  by a factor of $r^{-\ell_s}$.

  The radial function is a linear combination of
  $N_{\text{prim}}$ *primitive* functions. These primitives may be
  of Slater-type or Gaussian-type, and are parameterized by
  exponents $\gamma_{ks}$ and contraction coefficients $a_{ks}$.

  Slater orbitals are written as:

  \[

>   R_s^{\text{STO}}(\mathbf{r}) = \mathcal{N}_s  \left| \mathbf{r} - \mathbf{R}_A \right|^{n_s-1-\ell_s}

Reviewer: It's better to use the same convention as for Gaussians
```suggestion
  R_s^{\text{STO}}(\mathbf{r}) = \mathcal{N}_s  \left| \mathbf{r} - \mathbf{R}_A \right|^{n_s}
```

Author: OK, so then I will go back to a single expression for GTOs and STOs.

  spherical shell with $\ell_s = \lambda_s$ ($n_s=0$), $\ell_s =
  \lambda_s -2$ ($n_s=2$), $\ell_s = \lambda_s -4$ ($n_s=4$), etc.

  Standard basis sets assume that the primitive functions are
  normalized. In that case, a normalization constant $f_{ks}$ must
  be computed and applied to each primitive. This normalization
  constant usually depends on the exponent and on the quantum
  numbers $n$ and $\ell$. If this normalization is not required,
  then $f_{ks} = 1$.

  Similarly, some codes expect the contracted basis functions
  $R_s(\mathbf{r})$ to be normalized. In this case, an additional
  normalization factor $\mathcal{N}_s$ must be computed. If the
  full basis function is not normalized, then $\mathcal{N}_s = 1$.

*** Numerical orbitals

    TREXIO supports numerical atom centered orbitals. The implementation is
    based on the approach of FHI-aims [Blum, V. et al; Ab initio molecular
    simulations with numeric atom-centered orbitals; Computer Physics
    Communications 2009]. These orbitals are defined by the atom they
    are centered on, their angular momentum and a radial function
    $R_s$, which is of the form
    \[

>     R_s(\mathbf{r}) = \mathcal{N}_s \frac{u_i(\mathbf{r})}{r^{n_s}}.

Reviewer: I wonder about this expression since usually they define $R_s(r) = u_s(r)/r$ but here one also has the additional $r^{-\ell_s}$ term.

Author: Yes. I suppose the one who introduced this section was also confused with the original text of the GTO/STO... I am not familiar with this, I need to check.

Reviewer: did you get anywhere with this

* Basis functions
** Basis set (basis group)

  Different computational chemistry codes apply normalization at
  various levels. The normalization of basis functions requires the
  evaluation of diagonal overlap integrals, which might not always be
  supported by the client software. For this reason, normalization
  constants are explicitly provided in the TREXIO files to
  ensure portability and completeness.

  We consider here basis functions centered on nuclei. It is
  therefore possible to introduce *dummy atoms* (artificial centers
  with no physical meaning) solely for the purpose of placing basis
  functions at arbitrary locations in space.

  The atomic basis set is defined as a list of shells. Each shell

>   $s$ is centered on a point $\mathbf{R}_A$, associated with a given

Reviewer: the point is not associated with the angular momentum

```suggestion
  $s$ is centered on a point $\mathbf{R}_A$, it has a given
```

>   angular momentum $\ell_s$, and characterized by a radial function

Reviewer: ```suggestion
  angular momentum $\ell_s$, and is defined by a radial function
```

  \sum_{k=1}^{N_{\text{prim}}} f_{ks} \, a_{ks}\,
  \exp\left( -\gamma_{ks} || \mathbf{r} - \mathbf{R}_A ||^p \right)
  \]

  The radial exponent $n_s$ for STOs is connected to the primary quantum number
  $n$ of the shell by $n_s = n-1-\ell_s$.

  GTO-based codes usually use *spherically pure* Gaussians, in which
  case the radial prefactor exponent is $n_s = 0$, as the
  $r^{\ell_s}$ factor is already included in the solid harmonics.
  Similarly, *Cartesian Gaussians* correspond to $n_s = 0$,
  since the $r^{\lambda_s}$ term is already included in the
  Cartesian polynomial in the angular part.

  TREXIO is flexible enough to allow the implementation of Cartesian
  Gaussians in terms of spherical shells. For instance, a Cartesian
  $d$ shell consists of 6 functions, which are linear combinations of
  the 5 spherical d functions with $n_s=0$ and a spherical s
  function with $n_s=2$. More generally, a Cartesian shell of
  Cartesian angular momentum $\lambda_s$ will give rise to a
  spherical shell with $\ell_s = \lambda_s$ ($n_s=0$), $\ell_s =
  \lambda_s -2$ ($n_s=2$), $\ell_s = \lambda_s -4$ ($n_s=4$), etc.

>   Standard Gaussian basis sets, such as those available from the Basis
>   Set Exchange, typically assume that each /primitive/ function is
>   normalized.

Reviewer: ```suggestion
  Contracted Gaussian basis sets, such as those available from the Basis
  Set Exchange, are typically expressed in terms of normalized 
  /primitive/ functions.
```

  That is, for a primitive Gaussian of the form
  \[
  g_{ks}(\mathbf{r}) = ||\mathbf{r} - \mathbf{R}_k||^{\ell} \exp(-\alpha_s ||\mathbf{r} - \mathbf{R}_k||^2),
  \]
  a normalization constant $f_{ks}$ is computed so that the integral
  of $|f_{ks} g_{ks}|^2$ over all space is equal to 1.

>   It is applied independently for each primitive function.

Reviewer: ```suggestion
  Normalization is applied independently for each primitive function.
```

* Basis functions
** Basis set (basis group)

  Different computational chemistry codes apply normalization at
  various levels. The normalization of basis functions requires the
  evaluation of diagonal overlap integrals, which might not always be
  supported by the client software. For this reason, normalization
  constants should be explicitly provided in the TREXIO files to
  ensure portability and completeness.

> *** Gaussian- and Slater-type orbitals

Reviewer: why the change? this also applies to NAOs, no?

Author: Done

\sum_{k=1}^{N_{\text{prim}}}
f_{ks}\,a_{ks}\,
\exp\left( -\zeta_{ks} \left| \mathbf{r} - \mathbf{R}_A \right| \right)
    \]

    The prefactor exponent $n_s$ for STOs is $n_s = n-1-\ell_s$,
    where $n$ is the shell's principal quantum number.

    **Gaussian-type orbitals (GTO):**

    Gaussian-type orbitals (GTOs) are usually defined as:

    \[
    R_s^{\text{GTO}}(\mathbf{r}) = \mathcal{N}_s \left| \mathbf{r} - \mathbf{R}_A \right|^{n_s}
    \sum_{k=1}^{N_{\text{prim}}} f_{ks}^{\text{GTO}}(\gamma_{ks})\, a_{ks}\,
    R_s^{\text{GTO}}(\mathbf{r}) = \mathcal{N}_s
    \sum_{k=1}^{N_{\text{prim}}} f_{ks} \, a_{ks}\,
    \exp\left( -\gamma_{ks} \left| \mathbf{r} - \mathbf{R}_A \right|^2 \right)
    \]

    GTO-based codes usually use *spherically pure* Gaussians, in which
    case the radial prefactor exponent is \( n_s = 0 \), as the
    case the radial prefactor exponent is zero, as the
    $r^{\ell_s}$ factor is already included in the solid harmonics.
    Similarly, *Cartesian Gaussians* correspond to \( n_s = 0 \),
    since the $r^{\lambda_s}$ term is already included in the

>     Cartesian polynomial in the angular part.

Reviewer: this should not be removed

    TREXIO is flexible enough to allow the implementation of Cartesian
    Gaussians in terms of spherical shells. For instance, a Cartesian
    d shell consists of 6 functions, which are linear combinations of
    the 5 spherical d functions with $n_s=0$ and a spherical s
    function with $n_s=2$. More generally, a Cartesian shell of
    Cartesian angular momentum $\lambda_s$ will give rise to a
    spherical shell with $\ell_s = \lambda_s$ ($n_s=0$), $\ell_s =
    \lambda_s -2$ ($n_s=2$), $\ell_s =

>     \lambda_s -4$ ($n_s=4$), etc.

Reviewer: why remove this?
